README.md

```diff
@@ -50,20 +50,41 @@ _Note:_ The library exposes typings for IDE assistance/auto-complete and compile
 ### Initializing the client
 
 The client will use the OAuth2.0 flow `client_credentials` for authorization. Please refer to these [docs](https://developers.liveperson.com/liveperson-functions-external-invocations-client-credentials.html) for further information on that. On each request the client will check if the `JWT` is about to expire. If this is the case, the client will try to refresh it. If the `JWT` is expired and the client failed to refresh it, an `Error` is thrown. The time after which the refreshing logic will kick in can be specified via the property `jwtRefreshAfterMinutes`.
-Alternatively you can provide your own authorization method that generates a suitable authorization header.
 
 ```js
+// Option 1: Provide client id and client secret as auth strategy as follows
 import { Client } from 'liveperson-functions-client';
-// Either you provide client id and client secret as auth strategy as follows
+
 const client = new Client({
   accountId: 'myAccountId',
   authStrategy: {
     clientId: 'myClientId',
     clientSecret: 'myClientSecret',
   },
 });
-// Or you provide your own implementation of our type getAuthorizationHeader which looks as follows
-export type GetAuthorizationHeader = () => Promise<string>;
+```
+
+Alternatively you can provide your own authorization method that generates a suitable authorization header. To do so utilise our `GetAuthorizationHeader` type.
+
+How the type `GetAuthorizationHeader` looks like:
+```ts
+export type GetAuthorizationHeader = (input: {
+  url: string;
+  method: string;
+}) => Promise<string>;
+```
+Full example:
+```ts
+// Option 2: Provide your own implementation of our type getAuthorizationHeader
+
+import { Client, GetAuthorizationHeader } from 'liveperson-functions-client';
+
+const getAuthorizationHeader: GetAuthorizationHeader = async ({
+  url,
+  method,
+}) => {
+  // insert your implementation here
```

Comment:
Maybe show the OAuth as an example here ?

Reply:
implemented